modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureExceptionMatcher.java

```diff
@@ -119,4 +143,11 @@ public static CompletableFutureExceptionMatcher willThrow(Class<? extends Except
     public static CompletableFutureExceptionMatcher willThrow(Class<? extends Exception> cls, int timeout, TimeUnit timeUnit) {
         return willThrow(is(instanceOf(cls)), timeout, timeUnit);
     }
+
+    /**
+     * Creates a matcher that matches a future that completes with an exception that has a given {@code cause} in the exception stacktrace.
+     */
+    public static CompletableFutureExceptionMatcher willThrowWithCause(Class<? extends Exception> cause) {
```

Comment:
Should the method name (and javadoc) mention that the matcher also inspects suppressed exceptions? Like `willThrowWithCauseOrSuppressed()` or `willThrowWithRelatedException()`?

Reply:
I've copied this behavior from `IgniteUtils#assertThrowsWithCause`, so I think Ignite developers should be familiar with this method's name

Reply:
New developers will be quite surprised. Also, when a name is... not exactly telling truth, it's not good even if everyone is used to it. It's better to have truthful names :)